server/handlers.go

```diff
@@ -868,6 +898,16 @@ func (s *Server) handleRefreshToken(w http.ResponseWriter, r *http.Request, clie
 		scopes = requestedScopes
 	}
 
+	var connectorData []byte
+	if session, err := s.storage.GetOfflineSessions(refresh.Claims.UserID, refresh.ConnectorID); err != nil {
+		if err != storage.ErrNotFound {
+			s.logger.Errorf("failed to get offline session: %v", err)
+			return
+		}
+	} else {
+		connectorData = session.ConnectorData
+	}
+
```

Comment:
I guess to allow migration some code could be added here, checking if the `ConnectorData` is present, if not check `refresh` object for it. But this means putting the `ConnectorData` field back in the struct and would probably mean this migration code would be there forever.

I guess it depends how important you think maintaining existing refresh tokens is